subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/task/ToolchainReportRenderer.java

```diff
@@ -57,11 +59,41 @@ public void printInvalidToolchains(List<ReportableToolchain> invalidToolchains)
             for (ReportableToolchain toolchain : invalidToolchains) {
                 JvmInstallationMetadata metadata = toolchain.metadata;
                 output.withStyle(Identifier).println("     + " + metadata.getJavaHome());
-                final String paddedErrorType = Strings.padEnd("Error:", 20, ' ');
-                getTextOutput().withStyle(Normal).format("       | %s", paddedErrorType);
-                getTextOutput().withStyle(Description).println(metadata.getErrorMessage());
+                printInvalidToolchainErrorLines(toolchain);
             }
             output.println();
         }
     }
+
+    private void printInvalidToolchainErrorLines(ReportableToolchain invalidToolchain) {
+        List<Throwable> causeChain = new ArrayList<>();
+
+        Throwable cause = invalidToolchain.metadata.getErrorCause();
+        // Protect against excessively long cause-chains in the outputs.
+        // Try to get limit+1 elements to find out whether the limit is exceeded, see below
+        while (cause != null && causeChain.size() < INVALID_TOOLCHAIN_ERROR_CAUSE_LIMIT + 1) {
+            // Skip the first element, as the immediate error is already represented in metadata.getErrorMessage()
+            cause = cause.getCause();
+            if (cause != null) {
+                causeChain.add(cause);
+            }
+        }
+
+        List<String> errorLines = new ArrayList<>();
+        errorLines.add(invalidToolchain.metadata.getErrorMessage());
+        errorLines.addAll(causeChain.stream().map(Throwable::getMessage).limit(INVALID_TOOLCHAIN_ERROR_CAUSE_LIMIT).collect(Collectors.toList()));
+        if (causeChain.size() > INVALID_TOOLCHAIN_ERROR_CAUSE_LIMIT) {
+            // Ellipsize the omitted cause lines:
+            errorLines.add("...");
+        }
+
+        for (int i = 0; i < errorLines.size(); i++) {
+            String errorLinePrefix = Strings.padEnd(i == 0 ? "Error:" : "    Caused by:", 20, ' ');
```

Comment:
Given that we already format the first line differently - do we really need to collect all messages into a list first? Maybe we can print the first line separately and the rest - while iterating over the `causeChain`?

Reply:
Done, thanks!